libnetwork/netavark/config.go

```diff
@@ -38,6 +38,16 @@ func (n *netavarkNetwork) NetworkCreate(net types.Network) (types.Network, error
 }
 
 func (n *netavarkNetwork) networkCreate(newNetwork *types.Network, defaultNet bool) (*types.Network, error) {
+	if len(newNetwork.NetworkDNSServers) > 0 && !newNetwork.DNSEnabled {
+		return nil, fmt.Errorf("Cannot set NetworkDNSServers if DNS is not enabled for the network: %w", types.ErrInvalidArg)
+	} else if len(newNetwork.NetworkDNSServers) > 0 {
+		// validate ip address
+		for _, DNSServer := range newNetwork.NetworkDNSServers {
+			if net.ParseIP(DNSServer) == nil {
+				return nil, fmt.Errorf("Unable to parse ip %s specified in NetworkDNSServers: %w", DNSServer, types.ErrInvalidArg)
+			}
+		}
+	}
```

Comment:
this should not be the first step in the function, it has to be after
`internalutil.IpamNoneDisableDNS(newNetwork)`